### DISCUSSION TITLE
Virtual IP netmask

### DISCUSSION THREAD

**Discussion by liranodiz** [strongswan/strongswan#424] [General]
After reading: https://wiki.strongswan.org/projects/strongswan/wiki/VirtualIP
I am using virtual IP mode and successfully getting IP address by the "PLUTO_MY_SOURCEIP" environment variable.
Is it possible to get also the netmask of the address?

**Comment by tobiasbrunner** [comment]
No. strongSwan actually does not process the `INTERNAL_IP4|6_NETMASK` attributes at all. What's the reason you ask?

**Reply by liranodiz** [reply]
In the case of using strongSwan as initiator via responder that does not necessarily provide an address with netmask 32.

From RFC 5996:
![image](https://user-images.githubusercontent.com/24207367/120358289-002ff200-c30f-11eb-9a55-084a8437bb7a.png)

**Reply by tobiasbrunner** [reply]
And what practical use do you see in that? (By the way, RFC 7296 is the current one.)
